Isabelle/Chi_Calculus/Relaying.thy

```diff
@@ -13,32 +13,46 @@ lemma early_multi_receive_redundancy:
 
 lemma shortcut_redundancy:
   shows "a \<rightarrow> b \<parallel> b \<rightarrow> c \<parallel> a \<rightarrow> c \<approx>\<^sub>\<flat> a \<rightarrow> b \<parallel> b \<rightarrow> c"
-  by (fact early_multi_receive_redundancy)
+  using unidirectional_bridge_def and early_multi_receive_redundancy by simp
 
 lemma loop_redundancy_under_duploss:
   shows "\<currency>\<^sup>*a \<parallel> a \<rightarrow> a \<approx>\<^sub>\<flat> \<currency>\<^sup>*a"
   sorry
 
-abbreviation bidirectional_bridge :: "[chan, chan] \<Rightarrow> process" (infix "\<leftrightarrow>" 100) where
+definition bidirectional_bridge :: "[chan, chan] \<Rightarrow> process" (infix "\<leftrightarrow>" 100) where
   "a \<leftrightarrow> b \<equiv> a \<rightarrow> b \<parallel> b \<rightarrow> a"
 
 lemma bidirectional_bridge_commutativity: "a \<leftrightarrow> b \<sim>\<^sub>\<flat> b \<leftrightarrow> a"
-  by (simp add: parallel_commutativity)
+  by (simp add: parallel_commutativity bidirectional_bridge_def)
 
 lemma forward_bridge_absorption: "a \<leftrightarrow> b \<parallel> a \<rightarrow> b \<sim>\<^sub>\<flat> a \<leftrightarrow> b"
 proof -
   have "a \<leftrightarrow> b \<parallel> a \<rightarrow> b \<sim>\<^sub>\<flat> (a \<rightarrow> b \<parallel> a \<rightarrow> b) \<parallel> b \<rightarrow> a"
-    using basic.bisimilarity_transitivity_rule parallel_associativity parallel_commutativity by blast
+    using
+      unidirectional_bridge_def and
+      bidirectional_bridge_def and
+      basic.bisimilarity_transitivity_rule and
+      parallel_associativity parallel_commutativity
+    by metis
```

Comment:
Also here we should use `unfolding`, which would make the switch from `blast` to `metis` unnecessary.